src/duke/core/Duke.java

```diff
@@ -22,11 +22,18 @@
      * @param filePath
      */
     public Duke(String filePath) {
-
         database = new Database(filePath);
+    }
 
 
-
+    public ArrayList<Task> searchTask(ArrayList<Task> tasklist, String keyword) {
+        ArrayList<Task> result = new ArrayList<>();
+        for (Task task: tasklist) {
+            if (task.searchKeyword(keyword)) {
+                result.add(task);
+            }
+        }
+        return result;
```

Comment:
Consider adding JavaDoc comment on all of your implementations. It will help make it easier for people to understand what your code does, even if it seems obvious.